orchestra/utils/task_lifecycle.py

```diff
@@ -1019,6 +1036,41 @@ def end_project(project_id):
         notify_status_change(task, assignment_history(task))
 
 
+def set_project_status(project_id, status):
+    """
+    Set the project to the given status. Raise an exception if the status
+    is not a valid project status.
+
+    Args:
+        project_id (int): The ID of the project
+        status (str): The new project status
+
+    Returns:
+        None
+
+    Raises:
+        orchestra.core.errors.ProjectStatusError:
+            The specified new status is not supported.
+    """
+    project = Project.objects.get(id=project_id)
+    status_choices = dict(Project.STATUS_CHOICES)
+    if status == status_choices[Project.Status.PAUSED]:
+        project.status = Project.Status.PAUSED
+        slack_message = (
+            'This project is paused.')
+        message_experts_slack_group(project.slack_group_id, slack_message)
+    elif status == status_choices[Project.Status.ACTIVE]:
+        project.status = Project.Status.ACTIVE
+        slack_message = (
+            'The project is now active.')
+        message_experts_slack_group(project.slack_group_id, slack_message)
+    elif status == status_choices[Project.Status.ABORTED]:
+        project.status = Project.Status.ABORTED
```

Comment:
i think you probably want to bring in the logic for aborting projects, e.g, the end_project logic.